agent/taskresource/credentialspec/credentialspec_linux_test.go

```diff
@@ -154,7 +161,79 @@ func TestHandleSSMCredentialspecFile(t *testing.T) {
 	var wg sync.WaitGroup
 	errorEvents := make(chan error, len(cs.credentialSpecContainerMap))
 	wg.Add(1)
-	go cs.handleSSMCredentialspecFile(ssmCredentialSpec, credentialSpecSSMARN, iamCredentials, &wg, errorEvents)
+	go cs.handleSSMCredentialspecFile(ssmCredentialSpec, credentialSpecSSMARN, false, iamCredentials, &wg, errorEvents)
+
+	wg.Wait()
+	close(errorEvents)
+	err := <-errorEvents
+	assert.NoError(t, err)
+
+	cs.CredSpecMap[credentialSpecSSMARN] = expectedKerberosTicketPath
+
+	actualKerberosTicketPath, err := cs.GetTargetMapping(credentialSpecSSMARN)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedKerberosTicketPath, actualKerberosTicketPath)
```

Comment:
`cs.GetTargetMapping` method just gets the value from `cs.CredSpecMap`  which this test itself is setting on L171.  So, what is being tested here?

Reply:
this is just that map is properly populated

Reply:
I have a separate test that also covers the getmapping part, but this test it testing all the way from getting SSM info to the get mapping, I would remove it but I feel confident to have this assert ensuring in future nothing mess up the update mapping

Reply:
I see. Yeah I think we are mixing two test into one - 1) testing that `handleSSMCredentialspecFile` is able to call SSM client to download the data and parse it, and 2) `GetTargetMapping` is able to see updates made to `CredSpecMap` after  `handleSSMCredentialspecFile` has finished.

Maybe we can leave a comment explaining why the `GetTargetMapping` test is being performed in `TestHandleSSMDomainlessCredentialspecFile` as it is not clear just by looking at the test.

I still don’t see how `handleSSMCredentialspecFile` method is related to `CredSpecMap` though. Will leave this as a nit.

Reply:
we will take a look at it to see if that can be removed. Thanks Amogh